File: Testing/Web_Selenium/pages/AuthPage.js
const { By } = require('selenium-webdriver');
const BasePage = require('./BasePage');

class AuthPage extends BasePage {
    constructor(driver) {
        super(driver);
    }

    // Locators
    get emailInput() { return By.css('input[type="email"]'); }
    get passwordInput() { return By.css('input[type="password"]'); }
    get nameInput() { return By.css('input[placeholder="Enter your name"]'); }
    get submitButton() { return By.css('button[type="submit"]'); }
    get switchModeLink() { return By.xpath("//button[contains(text(), 'Sign Up') or contains(text(), 'Log In')]"); }
    get headerText() { return By.css('h2'); }
    get errorMessage() { return By.css('.bg-red-50 span'); } 
    get successMessage() { return By.css('.bg-emerald-50 span'); }

    // Methods
    async openAuth(mode = 'login') {
        if (mode === 'register') {
            await this.open('/auth?mode=register');
        } else {
            await this.open('/auth');
        }
    }

    async login(email, password) {
        await this.type(this.emailInput, email);
        await this.type(this.passwordInput, password);
        await this.click(this.submitButton);
    }

    async register(name, email, password) {
        await this.type(this.nameInput, name);
        await this.type(this.emailInput, email);
        await this.type(this.passwordInput, password);
        await this.click(this.submitButton);
    }
    
    async toggleMode() {
        await this.click(this.switchModeLink);
    }
}

module.exports = AuthPage;

File: qa-framework/selenium/pages/auth_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage

class AuthPage(BasePage):
    # Locators
    EMAIL_INPUT = (By.ID, "email")
    PASSWORD_INPUT = (By.ID, "password")
    NAME_INPUT = (By.ID, "name")
    SUBMIT_BTN = (By.XPATH, "//button[@type='submit']")
    TOGGLE_MODE = (By.XPATH, "//button[contains(text(), 'register') or contains(text(), 'login')]")
    FORGOT_PASSWORD_LINK = (By.XPATH, "//a[contains(text(), 'Forgot')]")
    ERROR_ALERT = (By.XPATH, "//*[contains(@class, 'bg-red-50') or contains(@class, 'text-red')]")

    def login(self, email, password):
        self.type(self.EMAIL_INPUT, email)
        self.type(self.PASSWORD_INPUT, password)
        self.click(self.SUBMIT_BTN)

    def register(self, name, email, password):
        # Switch to register mode if needed
        if not self.is_visible(self.NAME_INPUT):
            self.click(self.TOGGLE_MODE)
        self.type(self.NAME_INPUT, name)
        self.type(self.EMAIL_INPUT, email)
        self.type(self.PASSWORD_INPUT, password)
        self.click(self.SUBMIT_BTN)

    def get_error_message(self):
        if self.is_visible(self.ERROR_ALERT):
            return self.wait_for_element(self.ERROR_ALERT).text
        return None
